packages/console/src/pages/ConnectorDetails/index.tsx

```diff
@@ -28,6 +34,44 @@ const ConnectorDetails = () => {
   );
   const isLoading = !data && !error;
 
+  useEffect(() => {
+    if (data) {
+      setConfig(JSON.stringify(data.config, null, 2));
+    }
+  }, [data]);
+
+  const handleSave = async () => {
+    if (!connectorId) {
+      return;
+    }
+
+    if (!config) {
+      setSaveError(t('connector_details.save_error_empty_config'));
+
+      return;
+    }
+
+    try {
+      const configJson = JSON.parse(config) as JSON;
+      setIsSubmitLoading(true);
+      await ky
+        .patch(`/api/connectors/${connectorId}`, { json: { config: configJson } })
+        .json<ConnectorDTO>();
+      // TODO: show a toast, awaiting PR #354.
+    } catch (error: unknown) {
+      if (error instanceof SyntaxError) {
+        setSaveError(t('connector_details.save_error_json_parse_error'));
+      } else if (error instanceof HTTPError) {
+        const { message } = (await error.response.json()) as RequestErrorBody;
+        setSaveError(message);
+      } else {
+        console.error(error);
```

Comment:
we need to throw and handle general errors, maybe a `<ErrorBoundary>` component to show a toast?

can do it later, let's add an issue for it

Reply:
LOG-1868